PlutoHelperAgent/AppDelegate.m

```diff
@@ -193,7 +221,52 @@ -    (void)getUrl:(NSAppleEventDescriptor *)event
         BOOL extensionGood = [WhiteListProcessor checkIsInWhitelist:projectPath whiteListName:@"extensions_list" prefix:false];
               
         if (pathGood && extensionGood) {
-            [self tryOpenProject:projectPath];
+            NSArray *pathParts = [projectPath componentsSeparatedByString:@"?"];
```

Comment:
ooh, this is very hard to read.
Please can you split it up into its own functions so we can understand better what is going on.

I seem to remember reading, a long time back, that Linus Torvalds introduced a rule in the kernel that you were only allowed three levels of indentation - any more than that and you had to put in a function instead.  This was for readability; I am not suggesting a hard limit of 3, but we have a lot of nested if/else here that is very difficult to follow